### PR TITLE
DONTMERGE: explore running istio-iptables with user and group id 1337

### PR DESCRIPTION
**Pull request by lambdai**
Updated injection template generated golden files.
Exploring further mysteries.

Motivation: On top of https://github.com/istio/istio/pull/20130 and seeking simplified injection template

Ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/

### CODE REVIEW

File: install/kubernetes/helm/istio/files/injection-template.yaml [hunk 1]
```diff
@@ -44,19 +50,20 @@ initContainers:
...
       - NET_ADMIN
       - NET_RAW
+  {{- end }}
       drop:
       - ALL
-    privileged: {{ .Values.global.proxy.privileged }}
     readOnlyRootFilesystem: false
-    runAsGroup: 0
+    runAsGroup: 1337
     runAsNonRoot: false
```
Comment (rlenglet) [inline install/kubernetes/helm/istio/files/injection-template.yaml]: Should be `true`.

**Review by rlenglet — DISMISSED** [review]

**Comment by rlenglet** [comment]
Looking at `e2e-simpleTests_istio`:
```
2020-01-15T20:24:46.284173Z	info	Running command kubectl get pods -n istio-system --kubeconfig=
2020-01-15T20:24:46.373824Z	info	Command output: 
NAME                                      READY   STATUS                  RESTARTS   AGE
echosrv-deployment-1-5999458d77-dk77m     0/2     Init:CrashLoopBackOff   6          10m
echosrv-deployment-2-67f95f9f56-grvzk     0/2     Init:CrashLoopBackOff   6          10m
...
```
k8s is happy with the spec. However, it seems like the capabilities are not sufficient for `iptables`, cf. the [init container logs](https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/20186/e2e-simpleTests_istio/6165/artifacts/kind/istio-testing-control-plane/containers/echosrv-deployment-1-5999458d77-dk77m_istio-system_istio-init-afe996a722c2d677e29db03f3f1197bcf48a5ba1c7f52740b8962bb8604e238a.log):
```
2020-01-15T20:20:30.27289637Z stdout F iptables-restore --noflush /tmp/iptables-rules-1579119630272413513.txt304355156
2020-01-15T20:20:30.274762635Z stderr F iptables-restore v1.6.1: iptables-restore: unable to initialize table 'nat'
2020-01-15T20:20:30.27478649Z stderr F 
2020-01-15T20:20:30.274792787Z stderr F Error occurred at line: 1
2020-01-15T20:20:30.27480924Z stderr F Try `iptables-restore -h' or 'iptables-restore --help' for more information.
2020-01-15T20:20:30.275010565Z stdout F iptables-save 
2020-01-15T20:20:30.277898348Z stderr F Failed to list table names in /proc/net/ip_tables_names: Permission denied
2020-01-15T20:20:30.280543805Z stderr F panic: exit status 2
2020-01-15T20:20:30.280562141Z stderr F 
2020-01-15T20:20:30.280567818Z stderr F goroutine 1 [running]:
2020-01-15T20:20:30.280572662Z stderr F istio.io/istio/tools/istio-iptables/pkg/dependencies.(*RealDependencies).RunOrFail(0xd819a0, 0x97394b, 0x10, 0xc0000b8ba0, 0x2, 0x2)
2020-01-15T20:20:30.280584012Z stderr F 	istio.io/istio@/tools/istio-iptables/pkg/dependencies/implementation.go:44 +0x96
2020-01-15T20:20:30.280588878Z stderr F istio.io/istio/tools/istio-iptables/pkg/cmd.(*IptablesConfigurator).executeIptablesRestoreCommand(0xc000133d30, 0x7f4e93b48d01, 0x0, 0x0)
2020-01-15T20:20:30.280593633Z stderr F 	istio.io/istio@/tools/istio-iptables/pkg/cmd/run.go:474 +0x3aa
2020-01-15T20:20:30.280598818Z stderr F istio.io/istio/tools/istio-iptables/pkg/cmd.(*IptablesConfigurator).executeCommands(0xc000133d30)
2020-01-15T20:20:30.280603334Z stderr F 	istio.io/istio@/tools/istio-iptables/pkg/cmd/run.go:481 +0x45
2020-01-15T20:20:30.280607059Z stderr F istio.io/istio/tools/istio-iptables/pkg/cmd.(*IptablesConfigurator).run(0xc000133d30)
2020-01-15T20:20:30.280617873Z stderr F 	istio.io/istio@/tools/istio-iptables/pkg/cmd/run.go:428 +0x24e2
2020-01-15T20:20:30.280622088Z stderr F istio.io/istio/tools/istio-iptables/pkg/cmd.glob..func1(0xd5c720, 0xc000114700, 0x0, 0x10)
2020-01-15T20:20:30.280626871Z stderr F 	istio.io/istio@/tools/istio-iptables/pkg/cmd/root.go:56 +0x14e
2020-01-15T20:20:30.280709475Z stderr F github.com/spf13/cobra.(*Command).execute(0xd5c720, 0xc0000c8010, 0x10, 0x11, 0xd5c720, 0xc0000c8010)
2020-01-15T20:20:30.280714933Z stderr F 	github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
2020-01-15T20:20:30.28071965Z stderr F github.com/spf13/cobra.(*Command).ExecuteC(0xd5c720, 0x40574f, 0xc00009e058, 0x0)
2020-01-15T20:20:30.280724041Z stderr F 	github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
2020-01-15T20:20:30.28072806Z stderr F github.com/spf13/cobra.(*Command).Execute(...)
2020-01-15T20:20:30.280732169Z stderr F 	github.com/spf13/cobra@v0.0.5/command.go:864
2020-01-15T20:20:30.280736145Z stderr F istio.io/istio/tools/istio-iptables/pkg/cmd.Execute()
2020-01-15T20:20:30.280740222Z stderr F 	istio.io/istio@/tools/istio-iptables/pkg/cmd/root.go:277 +0x2d
2020-01-15T20:20:30.28074417Z stderr F main.main()
2020-01-15T20:20:30.280748215Z stderr F 	istio.io/istio@/tools/istio-iptables/main.go:22 +0x20
```
It seems like `iptables` needs to load kernel modules (`modprobe iptable_filter`) the first time it's used, and the container doesn't have this capability.

File: install/kubernetes/helm/istio/files/injection-template.yaml [hunk 2]
```diff
@@ -44,19 +50,20 @@ initContainers:
 {{- end }}
   securityContext:
     allowPrivilegeEscalation: {{ .Values.global.proxy.privileged }}
+    privileged: {{ .Values.global.proxy.privileged }}
     capabilities:
+  {{- if not .Values.istio_cni.enabled }}
       add:
       - NET_ADMIN
       - NET_RAW
```
Comment (rlenglet) [inline install/kubernetes/helm/istio/files/injection-template.yaml]: `iptables` needs to have the capability to load modules dynamically. Can you try to add `SYS_MODULE` explicitly?
```suggestion
       add:
       - NET_ADMIN
       - NET_RAW
       - SYS_MODULE
```

Comment (lambdai) [reply install/kubernetes/helm/istio/files/injection-template.yaml]: Thanks! Updating

Comment (lambdai) [reply install/kubernetes/helm/istio/files/injection-template.yaml]: TIL

File: manifests/istio-control/istio-autoinject/files/injection-template.yaml [hunk 3]
```diff
@@ -45,20 +51,21 @@ template: |
...
       readOnlyRootFilesystem: false
-      runAsGroup: 0
-      runAsNonRoot: false
-      runAsUser: 0
+      runAsGroup: 1337
+      runAsNonRoot: true
+      runAsUser: 1337
     restartPolicy: Always
-  {{- end }}
-  {{  end -}}
```
Comment (rlenglet) [inline manifests/istio-control/istio-autoinject/files/injection-template.yaml]: Don't remove this line.

**Comment by lambdai** [comment]
Bummer.
Will loop back later

**Pull request closed without merging**